lib/exercises.rb

```diff
@@ -1,19 +1,68 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) if you pretend line 12 is a simple mathematically based hashing function,
+# probably O (n * n log n) as for each string you have to sort the string to get a key for 
+# the hash. 
 
+# Space Complexity: O(n) - worst case is one string to one key, which is linearly dependent on 
+# the number of strings in the input array -- for an array of N strings, the has will at minimum expand
+# to hold N elements, and at most 2 * N elements
+
+# assume all anagrams are unique 
 def grouped_anagrams(strings)
```

Comment:
👍 The time complexity I think is O(n) if you assume each string is limited in size, like you say and O(n * m log m) where m is the max length of a string if they're not.